### PR TITLE
Remove default conversions

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/87006

```release-note
NONE
```

/kind cleanup
/priority important-longterm

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/conversion/converter.go [hunk 1]
```diff
@@ -192,117 +160,22 @@ func (c ConversionFuncs) Merge(other ConversionFuncs) ConversionFuncs {
 	return merged
 }
 
-// Meta is supplied by Scheme, when it calls Convert.
-type Meta struct {
-	// KeyNameMapping is an optional function which may map the listed key (field name)
-	// into a source and destination value.
-	KeyNameMapping FieldMappingFunc
-	// Context is an optional field that callers may use to pass info to conversion functions.
-	Context interface{}
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/conversion/converter.go:201]: don't drop meta and change the signature in this PR, just drop the reflective conversion bits that were already dead code

Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/conversion/converter.go]: done

**Comment by wojtek-t** [comment]
Thanks for the review - PTAL

**Pull request merged**